docs/source/optim.rst

```diff
@@ -4,11 +4,29 @@ Optim Module
 .. currentmodule:: cooper.optim
 
 
+.. _multiple-primal_optimizers:
+
+Multiple primal optimizers
+--------------------------
+
+When constructing a :py:class:`~cooper.constrained_optimizer.ConstrainedOptimizer`,
+one or multiple primal optimizers can be provided. When a list of optimizers is provided
+for the ``primal_optimizers`` argument, they are all treated as if they were a single
+optimizer. In particular, every operation applied to one optimizer is
+simultaneously applied to the other primal, without intermediate calls to
+:py:meth:`cmp.closure()<cooper.problem.ConstrainedMinimizationProblem.closure>` or
+:py:meth:`formulation.custom_backward(lagrangian)<cooper.problem.Formulation.custom_backward>`.
+
+Allowing for multiple primal optimizers is useful when setting different groups of
+primal variables to have different optimizer classes and hyperparameters.
```

Comment:
```suggestion
When constructing a :py:class:`~cooper.constrained_optimizer.ConstrainedOptimizer`,
one or multiple primal optimizers (grouped in a list) can be provided. Allowing 
for multiple primal optimizers is useful when setting separate groups of primal 
variables to have different optimizer classes and hyperparameters.

When a list of optimizers is provided for the ``primal_optimizers`` argument, they are
treated "as if they were a single optimizer". In particular, all primal optimizers 
operations such as :py:meth:`optimizer.step()<torch.optim.Optimizer.step>` are 
executed simultaneously (without intermediate calls to
:py:meth:`cmp.closure()<cooper.problem.ConstrainedMinimizationProblem.closure>` or
:py:meth:`formulation.custom_backward(lagrangian)<cooper.problem.Formulation.custom_backward>`).
```

Reply:
Also, maybe this should be moved *out* from the Optim Module docs and into the constrained optimizer's "Additional Features" section.

Reply:
I think users who will be interested in using multiple primal optimizers are likely familiar enough with Pytorch that "a list of optimizers" should be clear enough as a guide (along with the existing type hints and other docstrings). 
We probably do _not_ need to add a toy example demonstrating this feature.

Reply:
I agree that an example should not be neccessary. Moving to additional features sounds good.
Doing so and marking as resolved.